odl/tomo/backends/skimage_radon.py

```diff
@@ -9,104 +9,99 @@
-
-__all__ = ('skimage_radon_forward', 'skimage_radon_back_projector',
+__all__ = ('skimage_radon_forward_projector', 'skimage_radon_back_projector',
            'SKIMAGE_AVAILABLE')
 
 
-def skimage_theta(geometry):
-    """Calculate angles in degrees with ODL skimage conventions."""
-    return geometry.angles * 180.0 / np.pi
-
-
-def skimage_sinogram_space(geometry, volume_space, sinogram_space):
-    """Create a range adapted to the skimage radon geometry."""
+def skimage_proj_space(geometry, volume_space, proj_space):
+    """Create a projection space adapted to the skimage radon geometry."""
     padded_size = int(np.ceil(volume_space.shape[0] * np.sqrt(2)))
     det_width = volume_space.domain.extent[0] * np.sqrt(2)
-    skimage_detector_part = uniform_partition(-det_width / 2.0,
-                                              det_width / 2.0,
-                                              padded_size)
-
-    skimage_range_part = geometry.motion_partition.insert(
-        1, skimage_detector_part)
+    det_part = uniform_partition(-det_width / 2, det_width / 2, padded_size)
 
-    skimage_range = uniform_discr_frompartition(skimage_range_part,
-                                                interp=sinogram_space.interp,
-                                                dtype=sinogram_space.dtype)
+    part = geometry.motion_partition.insert(1, det_part)
+    space = uniform_discr_frompartition(
+        part, interp=proj_space.interp, dtype=proj_space.dtype
+    )
+    return space
 
-    return skimage_range
 
-
-def clamped_interpolation(skimage_range, sinogram):
+def clamped_interpolation(skimage_proj_space, sinogram):
     """Interpolate in a possibly smaller space.
 
     Sets all points that would be outside the domain to match the
     boundary values.
     """
-    min_x = skimage_range.domain.min()[1]
-    max_x = skimage_range.domain.max()[1]
+    min_x = skimage_proj_space.domain.min()[1]
+    max_x = skimage_proj_space.domain.max()[1]
 
-    def interpolation_wrapper(x):
+    def interpolator(x):
         x = (x[0], np.maximum(min_x, np.minimum(max_x, x[1])))
+        return sinogram.interpolation(x, bounds_check=False)
 
-        return sinogram.interpolation(x)
-    return interpolation_wrapper
+    return interpolator
 
 
-def skimage_radon_forward(volume, geometry, range, out=None):
+def skimage_radon_forward_projector(volume, geometry, proj_space, out=None):
     """Calculate forward projection using skimage.
 
     Parameters
     ----------
     volume : `DiscreteLpElement`
-        The volume to project
+        The volume to project.
     geometry : `Geometry`
-        The projection geometry to use
-    range : `DiscreteLp`
-        range of this projection (sinogram space)
-    out : ``range`` element, optional
-        An element in range that the result should be written to
+        The projection geometry to use.
+    proj_space : `DiscreteLp`
+        Space in which the projections (sinograms) live.
```

Comment:
Isn't this a bit too casually written?

Reply:
Well, it's not really a public function. Actually, I'll remove these implementation details from the `odl.tomo` namespace.